### PR TITLE
Add convenience constructors for Interval.

### PR DESCRIPTION
**Pull request by tkoolen**
I found these to be useful while writing the OSQP interface.

### CODE REVIEW

**Comment by odow** [comment]
Is this more suited to MOIU? I think we've tried to avoid convienence constructors like this in MOI.

**Comment by tkoolen** [comment]
I would consider that to be type piracy.

File: src/sets.jl [hunk 1]
```diff
@@ -102,6 +102,9 @@ struct Interval{T <: Real} <: AbstractScalarSet
     lower::T
     upper::T
 end
+Interval(s::GreaterThan) = Interval(s.lower, typemax(s.lower))
```
Comment (blegat) [inline src/sets.jl]: What do we do for `GreaterThan{Bool}` or `GreaterThan{Int}` ? In MathProgBase, we [throw a warning in that case](https://github.com/JuliaOpt/MathProgBase.jl/blob/7c6f0a8414fdc018cf7ba90eb3c5f29c2e68a738/src/HighLevelInterface/HighLevelInterface.jl#L6-L10).
We could also display a warning for non-`AbstractFloat` type or simply restrict this definition to `AbstractFloat` (I would prefer this second approach).

Comment (tkoolen) [reply src/sets.jl]: Good point. I also prefer the second approach.

**Comment by mlubin** [comment]
These need to be documented in some form. Either in the docstring for `Interval` or somewhere else.

**Comment by tkoolen** [comment]
Docs added.

**Pull request merged**